src/main/java/uk/gov/pay/connector/charge/service/ChargeService.java

```diff
@@ -247,12 +250,13 @@ private Optional<ChargeEntity> createCharge(ChargeCreateRequest chargeRequest, L
             var authorisationMode = chargeRequest.getAuthorisationMode();
 
             checkIfZeroAmountAllowed(chargeRequest.getAmount(), gatewayAccount);
-            if (!checkIfMotoApiAuthorisationModeIsAllowed(authorisationMode, gatewayAccount)) {
+            
+            if (!checkIfMotoApiAuthorisationModeIsAllowed(chargeRequest, gatewayAccount)) {
```

Comment:
I think this method is a bit confusing - although this was added earlier. From reading, its not immediately obvious what is being checked and when. I think something like the following would be clearer:

```suggestion
            if (chargeRequest.getAuthorisationMode() == MOTO_API) {
                checkMotoApiAuthorisationModeAllowed(gatewayAccount);
            } else {
                checkIfMotoPaymentsAllowed(chargeRequest.isMoto(), gatewayAccount);
            }
```

Reply:
fair enough, this can be made simpler now that the majority of the checks are done at the resource